src/toil/batchSystems/mesos/__init__.py

```diff
@@ -27,22 +31,110 @@
     'cores'))
 
 
-class ResourceRequirement( namedtuple('_ResourceRequirement', (
-    # Number of bytes (!) needed for a task
-    'memory',
-    # Number of CPU cores needed for a task
-    'cores',
-    # Number of bytes (!) needed for task on disk
-    'disk',
-    # True, if job can be run on a preemptable node, False otherwise
-    'preemptable'))):
+class JobQueue(object):
+
+    def __init__(self):
+        # mapping of jobTypes to queues of jobs of that type
+        self.queues = {}
+        # list of jobTypes in decreasing resource expense
+        self.sortedTypes = []
+
+    def insertJob(self, job, jobType):
+        # logn insertion
+        if jobType not in self.queues:
+            index = bisect(self.sortedTypes, jobType)
+            self.sortedTypes.insert(index, jobType)
+            self.queues[jobType] = Queue()
+
+        self.queues[jobType].put(job)
+
+    def sorted(self):
+        return self.sortedTypes
```

Comment:
`sortedTypes` can be modified by the leader thread while it's being iterated by the scheduler. I think this may not actually cause any problems right now (worst case a jobType is skipped temporarily or considered twice), but it might be nice to make this future-proof by doing `return list(self.sortedTypes)`, or at least noting the possibility of simultaneous modification in a comment in case this causes problems later.